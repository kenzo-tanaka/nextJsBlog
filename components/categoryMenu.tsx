import React from "react";
import Link from "next/link";
import { useRouter } from "next/router";
import { config } from "../site.config";

const CategoryMenu: React.FC = () => {
  const router = useRouter();
  const categoryList = config.categoryList;
  return (
    <div>
      {categoryList.map(({ slug, name }) => (
        <Link href={slug} key={slug}>
          <a
            className={`p-2 font-medium text-gray-800 w-full inline-block hover:bg-gray-100 hover:no-underline ${
              router.asPath === slug
                ? "border-l-4 border-gray-500 bg-gray-100"
                : ""
            }`}
          >
            {name}
          </a>
        </Link>
      ))}
    </div>
  );
};

export default CategoryMenu;
