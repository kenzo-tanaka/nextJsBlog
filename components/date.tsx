import { parseISO, format, parse } from "date-fns";

export default function Date({ dateString }) {
  const date = parseISO(dateString);
  return <time dateTime={dateString}>{format(date, "yyyy/LL/dd")}</time>;
}
