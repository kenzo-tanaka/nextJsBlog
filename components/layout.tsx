import Head from "next/head";
import styles from "./layout.module.css";
import utilStyles from "../styles/utils.module.css";
import Link from "next/link";
import { config } from "../site.config";

const Layout = ({ children, home = false }) => {
  return (
    <div className={styles.container}>
      <header className={styles.header}>
        {home ? (
          <>
            <img
              src="/images/profile.jpg"
              className={`${styles.headerHomeImage} ${utilStyles.borderCircle}`}
              alt={config.siteMeta.author}
            />
            <h1 className={utilStyles.heading2Xl}>{config.siteMeta.author}</h1>
          </>
        ) : (
          <>
            <Link href="/">
              <a>
                <img
                  src="/images/profile.jpg"
                  className={`${styles.headerImage} ${utilStyles.borderCircle}`}
                  alt={config.siteMeta.author}
                />
              </a>
            </Link>
            <h2 className={utilStyles.headingLg}>
              <Link href="/">
                <a className={utilStyles.colorInherit}>
                  {config.siteMeta.author}
                </a>
              </Link>
            </h2>
          </>
        )}
      </header>
      <main>{children}</main>
      {!home && (
        <div className={styles.backToHome}>
          <Link href="/">
            <a>‚Üê Back to home</a>
          </Link>
        </div>
      )}
    </div>
  );
};

export default Layout;
