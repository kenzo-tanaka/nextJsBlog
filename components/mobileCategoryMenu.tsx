import Link from "next/link";
import { useRouter } from "next/router";
import { config } from "../site.config";

const MobileCategoryMenu: React.FC = () => {
  const router = useRouter();
  const categoryList = config.categoryList;
  return (
    <div className="block sm:hidden grid grid-cols-3 mt-5 mb-8 text-center content-center">
      {categoryList.map(({ slug, name }) => (
        <Link href={slug} key={slug}>
          <a
            className={`m-auto font-medium text-gray-800 w-full pb-1 ${
              router.asPath === slug
                ? "border-b-4 border-gray-500 bg-gray-100"
                : ""
            }`}
          >
            {name}
          </a>
        </Link>
      ))}
    </div>
  );
};

export default MobileCategoryMenu;
