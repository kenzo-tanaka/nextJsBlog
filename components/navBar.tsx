import React from "react";
import Link from "next/link";
import Image from "next/image";

const NavBar: React.FC = () => {
  return (
    <nav className="bg-white border-b border-gray-200">
      <div className="max-w-4xl mx-auto px-2 sm:px-6 lg:px-8">
        <div className="relative flex items-center justify-between h-16">
          <div className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start">
            <div className="flex-shrink-0 flex items-center">
              <h1 className="font-bold font-sans text-xl text-gray-800">
                <Link href="/">{"{{ kenzo }}"}</Link>
              </h1>
            </div>
          </div>
          <div className="absolute inset-y-0 right-0 flex items-center pt-2 pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
            <Link href="/search">
              <a>
                <Image
                  src="/images/search.svg"
                  alt="GitHub Icon"
                  width={20}
                  height={20}
                />
              </a>
            </Link>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
