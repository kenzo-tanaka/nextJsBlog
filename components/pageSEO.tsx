import Head from "next/head";
import { config } from "../site.config";

type Props = {
  title: string;
  slug?: string;
  description?: string;
  ogImageUrl?: string;
};

export const pageSEO: React.FC<Props> = (props) => {
  const { title, slug, description, ogImageUrl } = props;
  const pageUrl = `${config.siteRoot}/${slug || ""}`;
  return (
    <Head>
      <title>{title}</title>
      <meta property="og:title" content={title} />
      <meta property="og:url" content={pageUrl} />
      <meta name="description" content={description} />
      <meta name="twitter:card" content="summary_large_image" />
      <meta property="og:site" content={config.siteMeta.title} />
      <meta
        property="og:image"
        content={ogImageUrl || `${config.siteRoot}/images/ogp.jpg`}
      />
    </Head>
  );
};
