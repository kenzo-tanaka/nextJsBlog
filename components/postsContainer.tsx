import Profile from "./profile";
import MobileCategoryMenu from "./mobileCategoryMenu";
import CategoryMenu from "./categoryMenu";
import PostsList from "./postsList";
import { PostData } from "@types";

type Props = {
  posts: PostData[];
};

const NavBar: React.FC<Props> = ({ posts }) => {
  return (
    <div className="shadow-md p-5">
      <Profile />
      <MobileCategoryMenu />
      <hr className="my-5 hidden sm:block" />
      <div className="grid grid-cols-6 gap-4">
        <div className="hidden md:block">
          <CategoryMenu />
        </div>
        <div className="col-span-6 md:col-span-5">
          <PostsList posts={posts} />
        </div>
      </div>
    </div>
  );
};

export default NavBar;
