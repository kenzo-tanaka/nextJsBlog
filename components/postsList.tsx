import { NextPage } from "next";
import Link from "next/link";
import utilStyles from "../styles/utils.module.css";
import Date from "./date";
import { PostData } from "@types";

type Props = {
  heading: string;
  posts: PostData[];
};

const PostsList: NextPage<Props> = ({ posts, heading }) => {
  return (
    <section className={`${utilStyles.headingMd} ${utilStyles.padding1px}`}>
      <h2 className={utilStyles.headingLg}>{heading}</h2>
      <ul className={utilStyles.list}>
        {posts.map(({ slug, date, title, category }) => (
          <li className={utilStyles.listItem} key={slug}>
            <Link href={`/posts/${slug}`}>
              <a className={`${utilStyles.textBold} ${utilStyles.titleText}`}>
                {title}
              </a>
            </Link>
            <br />
            <small className={utilStyles.lightText}>
              <Date dateString={date} />
            </small>
            <Link href={`/categories/${category}`}>
              <a className={utilStyles.categoryLabel}>#{category}</a>
            </Link>
          </li>
        ))}
      </ul>
    </section>
  );
};

export default PostsList;
