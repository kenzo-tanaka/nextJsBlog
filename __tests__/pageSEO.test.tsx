import React from "react";
import renderer from "react-test-renderer";
import { PageSEO } from "../components/pageSEO";

test("pageSEO", () => {
  const component = renderer.create(
    <PageSEO title="test-title" slug="test-slug" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
