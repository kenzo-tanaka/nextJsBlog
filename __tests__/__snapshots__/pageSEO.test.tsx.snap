// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pageSEO 1`] = `null`;
